libass/ass_utils.c

```diff
@@ -57,8 +57,41 @@ char *ass_strndup_fallback(const char *s, size_t n)
 #endif
 
 #ifndef HAVE_STRCHRNUL
+#if __has_attribute(no_sanitize)
+__attribute__((no_sanitize("address")))
+#endif
 char *ass_strchrnul_fallback(const char *s, int c)
 {
+#if __has_attribute(vector_size)
+
+#define VECTOR_SIZE 16
+
+#if __has_builtin(__builtin_ia32_pmovmskb128)
+#define ANY_ZERO(x) (__builtin_ia32_pmovmskb128((x) == 0) != 0)
+#elif __has_builtin(__builtin_neon_vminvq_u8) && 0
+#define ANY_ZERO(x) (__builtin_neon_vminvq_u8(x) == 0)
+#elif __has_builtin(__builtin_aarch64_reduc_umin_scal_v16qi_uu)
+#define ANY_ZERO(x) (__builtin_aarch64_reduc_umin_scal_v16qi_uu(x) == 0)
+#elif __has_builtin(__builtin_neon_vminvq_u8)
+#define ANY_ZERO(x) (__builtin_neon_vminvq_u8(x) == 0)
+#endif
+
+#endif // __has_attribute(vector_size)
+#ifdef ANY_ZERO
+    for (; (uintptr_t)s % VECTOR_SIZE; s++) {
+        if (!*s || *s == c)
+            return (char*)s;
+    }
+
+    typedef unsigned char CheckVector __attribute__ ((vector_size(VECTOR_SIZE)));
+
+    for (;; s += VECTOR_SIZE) {
+        CheckVector vec = *(CheckVector*)s;
+        if (ANY_ZERO(vec & ~(vec == (unsigned char)c)))
+            break;
+    }
+#endif // ANY_ZERO
```

Comment:
Maybe simple loop would be better here? Or even optimized intrinsics (for SSE2 only one `bsf` instruction left).

Reply:
I'd expect the impact to be negligible either way, and this was slightly simpler.